# 1 Assignment

It is totally prohibited to use any kind of loop. You can use stackoverflow. If you copy codes from previous answers, explain each step. No explanation is `0 points`.

## 1.1 Tuple



1. Print the first item of the second item of `tuple1` object. **Hint: Use indexing**<br><br>
2. Print the last item of the `tuple1` object.**Hint: Use indexing**<br><br>
3. Change the value "orange" for "pink grapefruit". Comment the output. **Hint: Is it possible?** <br><br>
4. Print the length of `tuple1`. **Hint: Length function**<br><br>
5. Sum all the elements of tuple2 and describe your steps and explain each of them. **Hint: Use `map` function.**<br><br>
6. Generate a new tuple object named as `tuple3` with the half values of `tuple2`. This tuple should be similar as `tuple2`, tuple of lists.  **Hint: Use `map` funciont.**<br><br>

In [22]:
tuple1 = ("Orange", [10, 20, 30], (5, 15, 25))
tuple2 = ([7, 8], [9, 1], [10, 7])

# 1.1.1 Print the first item of the second item of tuple1 object. Hint: Use indexing

In [4]:
tuple1 = ("Orange", [10, 20, 30], (5, 15, 25))
tuple1

('Orange', [10, 20, 30], (5, 15, 25))

Primero, se declara el objeto "tuple1" y se ejecuta para conocer los elementos que lo componen.

In [16]:
print(tuple1[1][0])

10


Para la ejecución de print, primero, se ubica el segundo item del objeto "tuple1". Este elemento es una lista "[10, 20, 30]" y está referenciado con [1]. Luego, se ubica el primer item de esta lista que es 10 y está referenciado con [0].
Finalmente queda: print(objeto = tuple1 [1] = segundo item [0] primer item del segundo item.

# 1.1.2. Print the last item of the tuple1 object. Hint: Use indexing

In [17]:
tuple1 = ("Orange", [10, 20, 30], (5, 15, 25))
tuple1

('Orange', [10, 20, 30], (5, 15, 25))

Se declara el objeto "tuple1" y se ejecuta para conocer los elementos que lo componen.

Para ejecutar print sobre el último elemento, existen dos opciones: Primero, se puede contar los elementos del tuple 1: "Orange", [10, 20, 30] y (5, 15, 25) y determinar que el último item está en el orden [2]. Por lo que el código quedaría así:

In [19]:
print(tuple1[2])

(5, 15, 25)


La segunda forma utiliza la indexación negativa. Para esto, se emplea [-1] para encontrar el último item de un objeto. En este caso tuple1[-1] muestra el último item de tuple1

In [20]:
print(tuple1[-1])

(5, 15, 25)


# 1.1.3. Change the value "orange" for "pink grapefruit". Comment the output. Hint: Is it possible?

En primera instancia, los "tuples" como objetos son inmutables. Es decir, sus elementos no pueden ser modificados una vez que han sido declarados. Sin embargo, se puede intentar recrear un tuple recreando los elementos de tuple1

In [23]:
tuple1 = ("pink grapefruit", tuple1[1], tuple1[2])
print(tuple1)

('pink grapefruit', [10, 20, 30], (5, 15, 25))


# 1.1.4. Print the length of tuple1. Hint: Length function

In [24]:
tuple1 = ("Orange", [10, 20, 30], (5, 15, 25))
tuple1

('Orange', [10, 20, 30], (5, 15, 25))

Primero, se declara el objeto "tuple1" y se ejecuta para conocer los elementos que lo componen.

In [25]:
print(len(tuple1))

3


Se utiliza print para "imprimir" el valor resultado de la función length (len) del objeto tuple1.

# 1.1.5. Sum all the elements of tuple2 and describe your steps and explain each of them. Hint: Use map function.

In [27]:
tuple2 = ([7, 8], [9, 1], [10, 7])
tuple2

([7, 8], [9, 1], [10, 7])

Primero, se declara el objeto "tuple2" y se ejecuta para conocer los elementos que lo componen. Los elementos son 3 listas: [7, 8] , [9, 1] y [10, 7]

In [42]:
suma_elementos = sum(map(sum, tuple2))
suma_elementos

42

Para sumar los elementos del objeto tuple2 se utiliza la función "sum", pero esta función no puede sumar listas, por lo que primero debemos sumar los elementos de cada lista y luego, sumar los resultados de cada lista. Para esto se utiliza la función "map" que aplica la función "sum" a todos los elementos de tuple2 "map(sum, tuple2)". Luego, los resultado de la función "map" son sumados mediante la función "sum": "sum(map(sum, tuple2)"

# 1.1.6 Generate a new tuple object named as tuple3 with the half values of tuple2. This tuple should be similar as tuple2, tuple of lists. Hint: Use map funciont

In [46]:
tuple2 = ([7, 8], [9, 1], [10, 7])
tuple2

([7, 8], [9, 1], [10, 7])

Primero, se declara el objeto "tuple2" y se ejecuta para conocer los elementos que lo componen

In [47]:
tuple3 = tuple(map(lambda sublist: [x / 2 for x in sublist], tuple2))
print(tuple3)

([3.5, 4.0], [4.5, 0.5], [5.0, 3.5])


Primero, se divide en 2 cada elemento de las listas que componen la tuple2. Para esto, se emplea la expresión lambda para dividir entre 2 (x/2 : 2):

lambda sublist: [x / 2 for x in sublist]

Luego, se aplica la función map para abarcar todos los valores divididos entre 2 (expresión lambda) de la tupla 2:

map(lambda sublist: [x / 2 for x in sublist], tuple2)

Por último, se declara el objeto tuple3 tomando los elementos del resultado de la función map:

tuple3 = tuple(map(lambda sublist: [x / 2 for x in sublist], tuple2))

Se ejecuta print de tuple 3 imprimiendo el resultado: ([3.5, 4.0], [4.5, 0.5], [5.0, 3.5])
